### PR TITLE
AC_AttitudeControl: always use smaller of slew yaw and rate max

### PR DESCRIPTION
**Pull request by IamPete1**
We had a error on line 581:
https://github.com/ArduPilot/ardupilot/blob/c8780792d4cf3fd7f5f2890f30ebe33536ce64cc/libraries/AC_AttitudeControl/AC_AttitudeControl.cpp#L571

This was throwing away max slew yaw because `_ang_vel_yaw_max` defaults to zero. The same bug as was fixed in https://github.com/ArduPilot/ardupilot/pull/18341

This adds some new methods to get max yaw rate and uses them in attitude control and copter auto yaw. Except for the bug fix above this should be no change in behavior.

### CODE REVIEW

File: libraries/AC_AttitudeControl/AC_AttitudeControl.cpp
Comment (rmackay9) [inline libraries/AC_AttitudeControl/AC_AttitudeControl.cpp:150]: maybe this comment should specify the units

File: libraries/AC_AttitudeControl/AC_AttitudeControl.cpp
Comment (rmackay9) [inline libraries/AC_AttitudeControl/AC_AttitudeControl.cpp:325]: sorry to be nitpicky, but let's put the "max" at the end of the variable name.

**Comment by tridge** [comment]
waiting for review from @lthall

**Comment by rmackay9** [comment]
If we could see some evidence of testing that would be great. thanks for this!

File: libraries/AC_AttitudeControl/AC_AttitudeControl.h
Comment (lthall) [inline libraries/AC_AttitudeControl/AC_AttitudeControl.h:120]: can we change this to:
get_ang_vel_yaw_max_degs

File: libraries/AC_AttitudeControl/AC_AttitudeControl.cpp
Comment (lthall) [reply libraries/AC_AttitudeControl/AC_AttitudeControl.cpp:325]: can we change this to:
get_ang_vel_yaw_max_degs

**Review by lthall — CHANGES_REQUESTED** [review]

File: libraries/AC_AttitudeControl/AC_AttitudeControl.cpp
Comment (IamPete1) [reply libraries/AC_AttitudeControl/AC_AttitudeControl.cpp]: done, both the comment and function name

File: libraries/AC_AttitudeControl/AC_AttitudeControl.cpp
Comment (IamPete1) [reply libraries/AC_AttitudeControl/AC_AttitudeControl.cpp]: Changed to `get_slew_ang_vel_yaw_max_rads` we already have a `get_ang_vel_yaw_max_rads`

File: libraries/AC_AttitudeControl/AC_AttitudeControl.h
Comment (IamPete1) [reply libraries/AC_AttitudeControl/AC_AttitudeControl.h]: changed to `get_slew_ang_vel_yaw_max_degs`

**Comment by IamPete1** [comment]
made changes, rebased and stepped through with GDB, works as expected.

File: libraries/AC_AttitudeControl/AC_AttitudeControl.h
Comment (lthall) [inline libraries/AC_AttitudeControl/AC_AttitudeControl.h:121]: get_slew_yaw_max_degs()

File: libraries/AC_AttitudeControl/AC_AttitudeControl.h
Comment (lthall) [inline libraries/AC_AttitudeControl/AC_AttitudeControl.h:376]: get_slew_yaw_max_rads()

File: libraries/AC_AttitudeControl/AC_AttitudeControl.cpp
Comment (lthall) [inline libraries/AC_AttitudeControl/AC_AttitudeControl.cpp:603]: slew_yaw_max_rads

File: libraries/AC_AttitudeControl/AC_AttitudeControl.cpp
Comment (lthall) [inline libraries/AC_AttitudeControl/AC_AttitudeControl.cpp:325]: slew_yaw_max_rads

**Review by lthall — CHANGES_REQUESTED** [review]
Sorry Pete, Naming variables is a pain. slew_ang_vel_yaw_max is double redundant with both slew and ang_vel.

Lets just go with:
get_slew_yaw_max_degs()
get_slew_yaw_max_rads()
slew_yaw_max_rads

I apologise for sending you around again.

**Review by lthall — APPROVED** [review]

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**
